ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -14,13 +14,615 @@
+	uint32_t byte_size;
+};
+
+/*
+ * SKS_CMD_CK_INIT_PIN - Initialiaze PKCS#11 token PIN
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [uint32_t pin_len]
+ *			   [uint8_t pin[pin_len]]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_InitPIN().
+ */
+#define SKS_CMD_CK_INIT_PIN		0x00000007
+
+struct sks_handle {
+	uint32_t handle;
+};
+
+/*
+ * SKS_CMD_CK_SET_PIN - Set PKCS#11 token PIN
+ *
+ * param#0: in(*)-memref : [uint32_t session_id]
+ *			   [uint32_t old_pin_len]
+ *			   [uint8_t old_pin[old_pin_len]]
+ *			   [uint32_t new_pin_len]
+ *			   [uint8_t new_pin[new_pin_len]]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_SetPIN()
+ */
+#define SKS_CMD_CK_SET_PIN		0x00000008
+
+/*
+ * SKS_CMD_CK_OPEN_RO_SESSION - Open Read-only Session
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [uint32_t session_handle]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_OpenSession() for a
+ * read-only session.
+ */
+#define SKS_CMD_CK_OPEN_RO_SESSION	0x00000009
+
+/*
+ * SKS_CMD_CK_OPEN_RW_SESSION - Open Read/Write Session
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [uint32_t session_handle]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_OpenSession() for a
+ * read/Write session.
+ */
+#define SKS_CMD_CK_OPEN_RW_SESSION	0x0000000a
+
+/*
+ * SKS_CMD_CK_CLOSE_SESSION - Open Read/Write Session
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_CloseSession().
+ */
+#define SKS_CMD_CK_CLOSE_SESSION	0x0000000b
+
+/*
+ * SKS_CMD_CK_SESSION_INFO - Get Cryptoki information on a session
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_session_info]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_GetSessionInfo().
+ */
+#define SKS_CMD_CK_SESSION_INFO		0x0000000c
+
+struct sks_ck_session_info {
+	uint32_t slot_id;
+	uint32_t state;
+	uint32_t flags;
+	uint32_t error_code;
+};
+
+/*
+ * SKS_CMD_CK_CLOSE_ALL_SESSIONS - Close all slot's pending sessions
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_CloseAllSessions().
+ */
+#define SKS_CMD_CK_CLOSE_ALL_SESSIONS	0x0000000d
+
+/*
+ * SKS_CMD_IMPORT_OBJECT - Open Read/Write Session
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [struct sks_object_head attribs + attributes data]
+ * param#1: none
+ * param#2: out-memref : [uint32_t object_handle]
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_ImportObject().
```

Comment:
AFAIK, there is no API C_ImportObject defined in PKCS#11 Standard.
There is an API, C_CreateObject for importing an already created object.
Please check.

Reply:
yes, true, my mistake. thank you.